Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp

```diff
@@ -70,6 +111,36 @@ bool iso_year_month_within_limits(i32 year, u8 month)
     return true;
 }
 
+// 9.5.5 BalanceISOYearMonth ( year, month ), https://tc39.es/proposal-temporal/#sec-temporal-balanceisoyearmonth
+ISOYearMonth balance_iso_year_month(double year, double month)
+{
+    // 1. Assert: year and month are integers.
+    VERIFY(year == trunc(year) && month == trunc(month));
+
+    // 2. Set year to year + floor((month - 1) / 12).
+    year += floor((month - 1) / 12);
+
+    // 3. Set month to (month − 1) modulo 12 + 1.
+    month = fmod(month - 1, 12) + 1;
+
+    // 4. Return the new Record { [[Year]]: year, [[Month]]: month }.
+    return ISOYearMonth { .year = static_cast<i32>(year), .month = static_cast<u8>(month), .reference_iso_day = 0 };
+}
+
+// 9.5.6 ConstrainISOYearMonth ( year, month ), https://tc39.es/proposal-temporal/#sec-temporal-constrainisoyearmonth
+ISOYearMonth constrain_iso_year_month(double year, double month)
+{
+    // 1. Assert: year and month are integers.
+    VERIFY(year == trunc(year) && month == trunc(month));
+
+    // 2. Set month to ! ConstrainToRange(month, 1, 12).
+    month = constrain_to_range(month, 1, 12);
+
+    // 3. Return the Record { [[Year]]: year, [[Month]]: month }.
+    // FIXME: What if year doesn't fit in an i32?!
```

Comment:
We can either verify that it does for now, or we can plumb global_object into here and do the same old `//IMPLEMENTATION DEFINED` thing, since afaict the only caller of this will immediately verify the year is valid (but we should say that so if it ever changes we can adapt)

Reply:
Done, this works nicely. Thanks!